command/s3_put.go

```diff
@@ -37,6 +37,9 @@ type s3put struct {
 	// included in this upload.
 	LocalFilesIncludeFilter []string `mapstructure:"local_files_include_filter" plugin:"expand"`
 
+	// LocalFilesIncludeFilterPrefix is an optional path to start processing the LocalFilesIncludeFilter, relative to the working directory.
+	LocalFilesIncludeFilterPrefix string `mapstructure:"local_files_include_filter_prefix" plugin:"expand"`
```

Comment:
Does this mean it doesn't automatically take the prefix as the part of the filter up until the first wildcard character?

Reply:
This constrains the working directory of the filter. For example, suppose you have something like:

<working_directory>
<working_directory>/foo
<working_directory>/bar

And suppose that you only want to apply a "local_files_include_filter" to files in <working_directory>/foo, because <working_directory>/bar contains a long path. You would set `local_files_include_filter_prefix: "foo"`.

Reply:
> Does this mean it doesn't automatically take the prefix as the part of the filter up until the first wildcard character?

I think what you're asking is whether setting `local_files_include_filter_prefix` changes `local_files_include_filter`. It does not. It only changes the working directory relative to which the `filepath.Walk` begins.

Reply:
I was suggesting that an include_filter of `foo/bar/**/*.java` or `foo/bar/*.xml` should imply an include_filter_prefix of `foo/bar` rather than have a separate parameter

Reply:
> I was suggesting that an include_filter of `foo/bar/**/*.java` or `foo/bar/*.xml` should imply an include_filter_prefix of foo/bar rather than have a separate parameter

Yeah, I think that's a better interface. Unfortunately, that requires implementing gitignore syntax as a preprocessor, which seemed like a pretty complicated implementation. I opted for a poorer interface to keep the implementation simple.